File: AEDS-II/Verde/Arquivo.c
#include <stdio.h>
#include <stdlib.h>

void LeitorDouble(FILE *arq, int qtdNumeros) {
    for (int i = 0; i < qtdNumeros; i++) {
        double n;
        scanf("%lf", &n);
        fwrite(&n, sizeof(double), 1, arq);
    }
    fclose(arq);
}

void LendoInverso(FILE *arq, int qtdNumeros) {
    for (int i = 0; i < qtdNumeros; i++) {
        fseek(arq, (qtdNumeros - i - 1) * sizeof(double), SEEK_SET);
        double valor;
        fread(&valor, sizeof(double), 1, arq);
        int coersao = (int) valor;
        if (coersao == valor) {
            printf("%d\n", coersao);
        } else if((valor * 1000 - (int) valor * 1000) == 0){
		printf("%.3f\n",valor);
	}
	else{
		printf("%g\n",valor);
	}
    }
    fclose(arq);
}

int main() {
    int qtdNumeros;
    scanf("%d", &qtdNumeros);
    FILE *arq = fopen("texto.txt", "wb");
    LeitorDouble(arq, qtdNumeros);
    arq = fopen("texto.txt", "rb");
    LendoInverso(arq, qtdNumeros);

    return 0;
}

File: AEDS-II/Verde/Arquivo.java
import java.io.IOException;
import java.io.RandomAccessFile;
class Arquivo{
	public static void LeitorDouble(RandomAccessFile arq, int qtdNumeros) throws IOException{
		for(int i = 0; i < qtdNumeros; i++){
			double n = MyIO.readDouble();
			arq.writeDouble(n);
		}
		arq.close();
	}
	public static void LendoInverso(RandomAccessFile arq, int qtdNumeros) throws IOException{
		for(int i = 0; i < qtdNumeros; i++){
			arq.seek((qtdNumeros - i - 1) * 8);
			double valor = arq.readDouble();
			int coersao = (int) valor;
			if(coersao == valor){
				MyIO.println(coersao);
			}
			else{
				MyIO.println(valor);
			}
		}
		arq.close();
	}
	public static void main(String [] args) throws IOException{
		int qtdNumeros = MyIO.readInt();
		RandomAccessFile arq = new RandomAccessFile("texto.txt", "rw");
		LeitorDouble(arq, qtdNumeros);
		arq = new RandomAccessFile("texto.txt","r");
		LendoInverso(arq, qtdNumeros);
	}
}
